### PR TITLE
Feat/list wrapper methods

### PR DESCRIPTION
**Pull request by phamvannhatvu**
- Implement a list wrapper with common methods: move/copy constructor, pushBack(), pushFront(), popBack(), popFront(), getSize(), empty(), begin(), end(), cbegin(), cend(), operator[]

### CODE REVIEW

File: ListWrapper.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#define LIST_WRAPPER_MIPS_SIMULATOR
+
+#include <queue>
+
+using std::deque;
+
+template<class T>
+Class ListWrapper {
+private:
+    deque<T> list;
```
Comment (HuyDNA) [inline ListWrapper.h:11]: Consider using vector instead for fast random access

**Review by HuyDNA — CHANGES_REQUESTED** [review]
Create a new folder called WrapperClasses and place this file there

File: WrapperClasses/ListWrapper.h [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    bool empty() const {
+        return list.empty();
+    }
+
+    ListWrapper() : list(vector<T>{}) {
+        
+    }
+
+    ListWrapper<T>(const ListWrapper<T> &wrapper) : list(vector<T>(wrapper.list)) {
```
Comment (HuyDNA) [inline WrapperClasses/ListWrapper.h:51]: Why the need to call vector<T>(...)?

File: WrapperClasses/ListWrapper.h [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    }
+
+    ListWrapper<T>(ListWrapper<T> &&wrapper) noexcept : list(wrapper.list) {
```
Comment (HuyDNA) [inline WrapperClasses/ListWrapper.h:55]: Should be std::move(wrapper.list)

**Review by HuyDNA — CHANGES_REQUESTED** [review]
Run test for this class to ensure it works properly

File: WrapperClasses/ListWrapper.h (hunk 2 above)
Comment (phamvannhatvu) [reply WrapperClasses/ListWrapper.h:51]: I thought merely list(wrapper.list) is a move assignment. I'll fix it soon!

**Review by HuyDNA — APPROVED** [review]

**Pull request merged**